sources/Renderer/RenderContext.cpp

```diff
@@ -133,6 +102,9 @@ void RenderContext::ShareSurfaceAndConfig(RenderContext& other)
     vsyncDesc_      = other.vsyncDesc_;
 }
 
+# if 0
+// XXX TODO On Mac and Windows platforms altering the user's display configuration
```

Comment:
Altering the user's display configuration works fine on Windows. Even when the application crashes, Windows resets the previous configuration. Only Linux has problems there when an application terminates abnormally. But for any rendering API on Windows other than D3D, you don't have much of a choice. Only D3D provides that functionality via the graphics API as far as I know.

Reply:
Altering the user's display configuration is frowned upon on the Mac and Apple does not believe it's necessary for performance reasons. And on Windows more apps are supporting borderless windowed fullscreen mode. In any case, the issue is that renderer back-ends can't opt out of this behavior. Even the D3D renderer's are not allowed to manipulate the display resolution using the D3D API calls.

Reply:
You're right, I see more and more games making use of a borderless window, too. If we can get this feature to work in LLGL, we can think about replacing the display configuration. That being said, I believe there are several philosophical design choices that are handled very differently between Mac and Windows. Most Windows games still provide the "true fullscreen" mode in which the user's display config is temporarily changed and I don't see the necessity (right now) to drop it immediately in LLGL. As for the Mac backend, this function was supposed to only move the App into fullscreen mode, without changing the display resolution or any other configuration, but iirc. it's not implemented yet.

Long story short, I'd prefer to handle the behavior of `SetDisplayMode` platform dependent. Whether it should be renamed then is another discussion.

Reply:
I believe both borderless fullscreen window and exclusive fullscreen are valid and LLGL should support both. Apple and Microsoft have written entire articles about how they've removed most of the bottlenecks related to borderless fullscreen mode. Still, I don't think LLGL can get rid of exclusive fullscreen so I'm thinking about how to structure the API to support both.

As for D3D, neither of the backends seem to be manipulating the display resolution using the Direct3D interfaces (e.g. ResizeTarget). Is that intentional? Bear in mind that I may be misunderstanding the Microsoft documentation.